src/pages/sign-in/index.tsx

```diff
@@ -50,30 +84,23 @@ const Index = () => {
             Sign in to your account
           </span>
         </div>
-        <div className="w-full">
-          <SignInTextField
-            placeholder={"Email"}
-            type={"text"}
-            dataCy={"sign-in-email"}
-            onChange={(e: React.ChangeEvent<HTMLInputElement>) => setEmail(e.target.value)}
-          />
-        </div>
-        <div className="w-full">
-          <SignInTextField
-            placeholder={"Password"}
-            type={"password"}
-            dataCy={"sign-in-password"}
-            onChange={(e: React.ChangeEvent<HTMLInputElement>) => setPassword(e.target.value)}
-          />
-        </div>
-        {showErrorPopup && (
-          <ErrorPopup
-            closePopup={() => setErrorPopup(false)}
-            text={errorPopupText}
-            dataCy={"error-popup-sign-in"}
-          />
-        )}
-        <MainButton onClick={signIn} text={"Sign in"} dataCy={"sign-in-submit-btn"} />
+        {signInFormData.map(({ dataCy, placeholder, validationOptions, tag, type }) => {
+          return (
+            <div className="flex w-full flex-col gap-2" key={tag}>
+              <SignInTextField
+                tag={tag}
+                validationOptions={validationOptions}
+                register={register}
+                type={type}
+                placeholder={placeholder}
+                dataCy={dataCy}
+              />
+              {errors[tag] && <ErrorMsg dataCy={`${dataCy}-error`} message={errors[tag]?.message?.toString()} />}
+            </div>
+          )
+        })}
+        {showErrorMsg && <ErrorMsg dataCy={"sign-in-err"} message={errorPopupText} />}
+        <MainButton onClick={handleSubmit(signIn)} text={"Sign in"} dataCy={"sign-in-submit-btn"} />
```

Comment:
If you put this in a `<form>` you can set the type of this button to 'submit' and set the onSubmit of `<form>` to `handleSubmit(signIn)`

Also since MainButton has no other function but apply the className, I'm not sure whether it is worth it to keep it in a separate component. You could just create a custom class (named `btn` for example) for it in [src/styles/globals.css](https://github.com/ASVGay/the-rhapsodies/blob/88871f442a7b1d967f7105c47a8df527d79c89ef/src/styles/globals.css#L20) and use the class whenever you're creating a button. Also allows for more customization while still keeping the base styling.